app/assets/javascripts/comments.js

```diff
@@ -1,5 +1,46 @@
 $( function () {
-  $('.edit_comment #comment_content').live('change', RescueRails.saveParentForm);
+
+  if ( $('.edit_comment').length > 0) {
+    $('.toggle-edit-comment').each(function(){
+      var isEditing = false;
+
+      $(this).click(function() {
+
+        var $parent = $(event.target).parents('.edit-comment')
+
+        if (isEditing) {
+          $parent.find('.read-only-comment').show();
+          $parent.find('.editable-comment').hide();
+          $parent.find('.toggle-edit-comment').addClass('btn-primary').text('Edit');
+          $parent.find('.save-edit-comment').hide();
+        }
+        else {
+          $parent.find('.read-only-comment').hide();
+          $parent.find('.editable-comment').show();
+          $parent.find('.toggle-edit-comment').removeClass('btn-primary').text('Cancel');
+          $parent.find('.save-edit-comment').show();
+        }
+        isEditing = !isEditing;
+
+      });
+
+      $('.save-edit-comment').click(function() {
```

Comment:
this whole block is being executed for every comment, so when you click a save button it will be executed comment # of times, you'll need to move it out and as a result the $parent.find stuff should get their own functions so you can shut off edit mode from the save function.

```javascript
function editComment($parent) {
  $parent.find('.read-only-comment').hide();
  $parent.find('.editable-comment').show();
  $parent.find('.toggle-edit-comment').removeClass('btn-primary').text('Cancel');
  $parent.find('.save-edit-comment').show();
}
```